--- xen-4.4.0/tools/python/Makefile.orig	2014-10-08 10:02:06.644246630 +0100
+++ xen-4.4.0/tools/python/Makefile	2014-10-08 10:02:19.445841687 +0100
@@ -16,11 +16,11 @@
 		$(XEN_ROOT)/tools/libxl/libxl_types.idl \
 		xen/lowlevel/xl/_pyxl_types.h \
 		xen/lowlevel/xl/_pyxl_types.c
-	CC="$(CC)" CFLAGS="$(CFLAGS)" $(PYTHON) setup.py build --xend=$(CONFIG_XEND)
+	CC=clang CFLAGS="$(CFLAGS)" $(PYTHON) setup.py build --xend=$(CONFIG_XEND)
 
 .PHONY: install
 install: install-dtd
-	CC="$(CC)" CFLAGS="$(CFLAGS)" $(PYTHON) setup.py install \
+	CC=clang CFLAGS="$(CFLAGS)" $(PYTHON) setup.py install \
 		$(PYTHON_PREFIX_ARG) --root="$(DESTDIR)" --force --xend=$(CONFIG_XEND)
 
 	$(INSTALL_DIR) $(DESTDIR)$(SBINDIR)
--- xen-4.4.0/tools/python/setup.py.orig	2014-10-08 10:20:21.958128928 +0100
+++ xen-4.4.0/tools/python/setup.py	2014-10-08 10:20:34.041128037 +0100
@@ -4,7 +4,7 @@
 
 XEN_ROOT = "../.."
 
-extra_compile_args  = [ "-fno-strict-aliasing", "-Werror" ]
+extra_compile_args  = [ "-fno-strict-aliasing" ]
 
 PATH_XEN      = XEN_ROOT + "/tools/include"
 PATH_LIBXC    = XEN_ROOT + "/tools/libxc"
--- xen-4.4.0/tools/pygrub/Makefile.orig	2014-10-08 10:33:04.675072736 +0100
+++ xen-4.4.0/tools/pygrub/Makefile	2014-10-08 10:33:20.251071588 +0100
@@ -6,11 +6,11 @@
 all: build
 .PHONY: build
 build:
-	CC="$(CC)" CFLAGS="$(CFLAGS)" $(PYTHON) setup.py build
+	CC=clang CFLAGS="$(CFLAGS)" $(PYTHON) setup.py build
 
 .PHONY: install
 install: all
-	CC="$(CC)" CFLAGS="$(CFLAGS)" $(PYTHON) setup.py install \
+	CC=clang CFLAGS="$(CFLAGS)" $(PYTHON) setup.py install \
 		$(PYTHON_PREFIX_ARG) --root="$(DESTDIR)" \
 		--install-scripts=$(PRIVATE_BINDIR) --force
 	$(INSTALL_DIR) $(DESTDIR)/var/run/xend/boot
--- xen-4.4.0/tools/pygrub/setup.py.orig	2014-10-08 10:33:27.274071071 +0100
+++ xen-4.4.0/tools/pygrub/setup.py	2014-10-08 10:33:35.613070457 +0100
@@ -3,7 +3,7 @@
 import os
 import sys
 
-extra_compile_args  = [ "-fno-strict-aliasing", "-Werror" ]
+extra_compile_args  = [ "-fno-strict-aliasing" ]
 
 XEN_ROOT = "../.."
 
